common/src/attestations/signer.rs

```diff
@@ -102,21 +172,155 @@ pub struct Attestation {
     pub s: Bytes32,
 }
 
-/// Helper for creating an AttestationSigner
-pub fn create_attestation_signer(
-    chain_id: U256,
-    dispute_manager_address: Address,
-    signer: SigningKey,
-    deployment_id: DeploymentId,
-) -> anyhow::Result<AttestationSigner> {
-    // Tedious conversions to the "indexer_native" types
-    let mut chain_id_bytes = [0u8; 32];
-    chain_id.to_big_endian(&mut chain_id_bytes);
-    let signer = AttestationSigner::new(
-        eip_712_derive::U256(chain_id_bytes),
-        dispute_manager_address,
-        secp256k1::SecretKey::from_slice(&signer.to_bytes())?,
-        deployment_id,
-    );
-    Ok(signer)
+fn wallet_for_allocation(
+    indexer_mnemonic: &str,
+    allocation: &Allocation,
+) -> Result<Wallet<SigningKey>, anyhow::Error> {
+    // Guess the allocation index by enumerating all indexes in the
+    // range [0, 100] and checking for a match
+    for i in 0..100 {
```

Comment:
Is the allocation index search just a remnant from when we could open parallel allocations? Is it still needed?

Reply:
Yes, exactly! I think opening parallel allocations is still possible and probably still done here and there?

Reply:
I know that it's not supported in the indexer-agent anymore (and that's been for quite a while now IIRC), so supporting it in the indexer-service would only be useful for indexers that use some unsupported method of opening their allocations.
Do you think it'd make sense to propose deprecating it in the indexer-service?

Reply:
Let's do that separately. I agree it may be worth doing but it doesn't hurt to keep it in for the moment.

Reply:
:100: